pkg/skaffold/build/parallel.go

```diff
@@ -50,21 +50,25 @@ func InParallel(ctx context.Context, out io.Writer, tags tag.ImageTags, artifact
 		return runInSequence(ctx, out, tags, artifacts, buildArtifact)
 	}
 
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	results := new(sync.Map)
 	outputs := make([]chan []byte, len(artifacts))
 
 	// Run builds in //
+	wg.Add(len(artifacts))
 	for i := range artifacts {
 		outputs[i] = make(chan []byte, buffSize)
 		r, w := io.Pipe()
 		cw := setUpColorWriter(w, out)
 
 		// Run build and write output/logs to piped writer and store build result in
 		// sync.Map
-		go runBuild(ctx, cw, tags, artifacts[i], results, buildArtifact)
+		go runBuild(ctx, &wg, cw, tags, artifacts[i], results, buildArtifact)
```

Comment:
If I put a closure here to avoid passing in `wg`, it would cause tests to fail for me by changing the build order

Reply:
This is because `i` needs to be captured with something like `i := i` before running the go routine. I'd rather we don't pass the waitGroup as an argument to runBuild. It makes the code harder to reason about.

Reply:
Ah, thank you!